Sync 'RTCRtpScriptTransform.idl' with WebIDL Specification

youennf opened the pull request:
#### 91ff5d82ae24524495857a0097aede353ffa00f4
<pre>
Sync &apos;RTCRtpScriptTransform.idl&apos; with WebIDL Specification
<a href="https://bugs.webkit.org/show_bug.cgi?id=264905">https://bugs.webkit.org/show_bug.cgi?id=264905</a>
<a href="https://rdar.apple.com/118728503">rdar://118728503</a>

Reviewed by Jean-Yves Avenard.

Update IDL definition according spec.
Remove EventTarget RTCRtpScriptTransform related code.

Add a test for the case where options is not set.
Resync WPT webrtc-encoded-transform WebIDL tests.

* LayoutTests/http/wpt/webrtc/no-options-transform-worker.js: Added.
(onrtctransform):
* LayoutTests/http/wpt/webrtc/script-transform-no-options-expected.txt: Added.
* LayoutTests/http/wpt/webrtc/script-transform-no-options.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/interfaces/webrtc-encoded-transform.idl:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window.js: Added.
(async idlArray):
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransform.h:
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransform.idl:
* Source/WebCore/dom/EventTargetFactory.in:

Canonical link: <a href="https://commits.webkit.org/271380@main">https://commits.webkit.org/271380@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14eefa89974eaaa649823b0f8547210a9b436035

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/28209 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/6850 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/29562 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/30738 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/25703 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/28706 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/8887 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/4233 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/25988 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/28477 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/5620 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/24281 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/4891 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5018 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/25277 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/31427 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/25826 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/25708 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/31329 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/4997 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/3176 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29084 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/6545 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/25077 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6754 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/5434 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/5512 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard commented on LayoutTests/imported/w3c/web-platform-tests/interfaces/webrtc-encoded-transform.idl:
```diff
@@ -48,40 +64,42 @@ enum RTCEncodedVideoFrameType {
+    unsigned long synchronizationSource;
+    octet payloadType;
+    sequence<unsigned long> contributingSources;
+    long long timestamp;    // microseconds
```
no rtpTimestamp or mimeType attribute?

jyavenard commented on LayoutTests/imported/w3c/web-platform-tests/interfaces/webrtc-encoded-transform.idl:
```diff
@@ -48,40 +64,42 @@ enum RTCEncodedVideoFrameType {
 interface RTCEncodedVideoFrame {
     readonly attribute RTCEncodedVideoFrameType type;
-    readonly attribute unsigned long long timestamp;
+    readonly attribute unsigned long timestamp;
```
this isn't in https://w3c.github.io/webrtc-encoded-transform/#RTCEncodedVideoFrame-interface

jyavenard commented on LayoutTests/imported/w3c/web-platform-tests/interfaces/webrtc-encoded-transform.idl:
```diff
@@ -48,40 +64,42 @@ enum RTCEncodedVideoFrameType {
+    unsigned long synchronizationSource;
+    octet payloadType;
+    sequence<unsigned long> contributingSources;
+    short sequenceNumber;
```
no rtpTimestamp nor mimeType ?

jyavenard reviewed (commented):
is there an advantage in syncing the webIDL files only?

no related implementation?

youennf committed:
Sync 'RTCRtpScriptTransform.idl' with WebIDL Specification
https://bugs.webkit.org/show_bug.cgi?id=264905
rdar://118728503

Reviewed by Jean-Yves Avenard.

Update IDL definition according spec.
Remove EventTarget RTCRtpScriptTransform related code.

Add a test for the case where options is not set.
Resync WPT webrtc-encoded-transform WebIDL tests.

* LayoutTests/http/wpt/webrtc/no-options-transform-worker.js: Added.
(onrtctransform):
* LayoutTests/http/wpt/webrtc/script-transform-no-options-expected.txt: Added.
* LayoutTests/http/wpt/webrtc/script-transform-no-options.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/interfaces/webrtc-encoded-transform.idl:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window.js: Added.
(async idlArray):
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransform.h:
* Source/WebCore/Modules/mediastream/RTCRtpScriptTransform.idl:
* Source/WebCore/dom/EventTargetFactory.in:

Canonical link: https://commits.webkit.org/271380@main
